src/poliastro/plotting.py

```diff
@@ -394,6 +394,226 @@ def savefig(self, filename, **layout_kwargs):
+        self._attractor_data = {}  # type: dict
+        self._attractor_radius = np.inf * u.km
+
+        self._color_cycle = cycle(plotly.colors.DEFAULT_PLOTLY_COLORS)
+
+    @property
+    def figure(self):
+        return dict(
+            data=self._data + [self._attractor_data],
+            layout=self._layout,
+        )
+
+    def _plot_circle(self, radius, color, name, center=[0, 0, 0] * u.km):
+        xx, yy = _generate_circle(radius, center)
+        x_center, y_center, z_center = center
+        trace = Scatter(x=xx.to(u.km).value, y=yy.to(u.km).value, mode='markers', line=dict(color=color, width=5,
+                                                                                            dash='dash',), name=name)
+        self._layout["shapes"].append(
+            {
+                'type': 'circle',
+                'xref': 'x',
+                'yref': 'y',
+                'x0': (x_center - radius).to(u.km).value,
+                'y0': (y_center - radius).to(u.km).value,
+                'x1': (x_center + radius).to(u.km).value,
+                'y1': (y_center + radius).to(u.km).value,
+                'opacity': 1,
+                'fillcolor': color,
+                'line': {
+                    'color': color,
+                },
+            },
+
+        )
+        return trace
+
+    def _redraw_attractor(self, min_radius=0 * u.km):
+        # Select a sensible value for the radius: realistic for low orbits,
+        # visible for high and very high orbits
+        radius = max(self._attractor.R.to(u.km), min_radius.to(u.km))
+
+        # If the resulting radius is smaller than the current one, redraw it
+        if radius < self._attractor_radius:
+            circle = self._plot_circle(
+                radius, BODY_COLORS.get(self._attractor.name, "#999999"), self._attractor.name)
+
+            # Overwrite stored properties
+            self._attractor_radius = radius
+            self._attractor_data = circle
+
+    def _plot_trajectory(self, trajectory, label, color, dashed):
+        trace = Scatter(
+            x=trajectory.x.to(u.km).value, y=trajectory.y.to(u.km).value,
+            name=label,
+            line=dict(
+                color=color,
+                width=2,
+                dash='dash' if dashed else 'solid',
+            ),
+            mode="lines",  # Boilerplate
+        )
+        self._data.append(trace)
+
+    def set_attractor(self, attractor):
+        """Sets plotting attractor.
+
+        Parameters
+        ----------
+        attractor : ~poliastro.bodies.Body
+            Central body.
+
+        """
+        if self._attractor is None:
+            self._attractor = attractor
+
+        elif attractor is not self._attractor:
+            raise NotImplementedError("Attractor has already been set to {}.".format(self._attractor.name))
+
+    @u.quantity_input(elev=u.rad, azim=u.rad, distance=u.km)
+    def set_view(self, elev, azim, distance=5 * u.km):
```

Comment:
Does this make sense for 2D plots?

Reply:
No! But as we plan to further merge the 3D and 2D classes, so I kept it. I will remove it!